src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -105,6 +109,46 @@ compileDotProtoFileOrDie args = compileDotProtoFile args >>= \case
     |]
   _ -> pure ()
 
+-- | Renaming protobuf file names to valid Haskell module names.
+--
+-- By convention, protobuf filenames are snake case. 'rnProtoFile' renames
+-- snake-cased protobuf filenames by:
+--
+-- * Replacing occurrences of one or more underscores followed by an
+-- alphabetical character with one less underscore.
+--
+-- * Capitalizing the first character following the string of underscores.
+--
+-- ==== __Examples__
+--
+-- >>> rnProtoFile @(Either CompileError) "abc_xyz"
+-- Right "AbcXyz"
+--
+-- >>> rnProtoFile @(Either CompileError) "abc_1bc"
+-- Left (InvalidModuleName "_1bc")
+--
+-- >>> rnProtoFile @(Either CompileError) "_"
+-- Left (InvalidModuleName "_")
+rnProtoFile :: MonadError CompileError m => String -> m String
+rnProtoFile str = do
```

Comment:
Unless you're copying an idiom from somewhere else, I'd call this `renameProtoFile` instead of `rnProtoFile`.